core/src/main/java/com/linecorp/armeria/server/encoding/HttpEncoders.java

```diff
@@ -76,6 +77,8 @@ static OutputStream getEncodingOutputStream(HttpEncodingType encodingType, Outpu
                     throw new IllegalStateException(
                             "Error writing brotli header. This should not happen with byte arrays.", e);
                 }
+            case SNAPPY:
+                return new SnappyOutputStream(out);
```

Comment:
I guess maybe we should use `SnappyFramedOutputStream` instead.

ref: https://github.com/xerial/snappy-java#compatibility-notes

Reply:
Thanks for the feedback. Will update that and I think I'm also missing if condition that returns HttpEncodingType.SNAPPY in lines 116 and 138. Will add those as well.

Reply:
Researched the framing format. After reading the official documentation, I agree that `SnappyFramedOutputStream` is a better choice for HTTP not to hold the entire buffer.
https://github.com/google/snappy/blob/27f34a580be4a3becf5f8c0cba13433f53c21337/docs/README.md?plain=1#L18-L21
There is no official encoding type but `x-snappy-framed` is recommended for the framed type.
https://github.com/google/snappy/blob/27f34a580be4a3becf5f8c0cba13433f53c21337/framing_format.txt#L35-L36C27